Add user and location repeaters for BETS

esoergel opened the pull request:
Built on top of https://github.com/dimagi/commcare-hq/pull/15957

NoahCarnahan committed:
Add files

NoahCarnahan committed:
Add BETS toggle

NoahCarnahan committed:
Move case_properties_changed() to utils file

NoahCarnahan committed:
Add base classes

NoahCarnahan committed:
Add BETSVoucherRepeater

NoahCarnahan committed:
Add other repeaters

NoahCarnahan committed:
Add tests

NoahCarnahan committed:
Combine const files

NoahCarnahan committed:
Add test_payload functions

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into nc/bets

NoahCarnahan committed:
lint

NoahCarnahan committed:
Remove legacy signal

NoahCarnahan committed:
lint

NoahCarnahan committed:
Add BETS repeaters to enikshay repeater filter

NoahCarnahan committed:
Fix indentation

NoahCarnahan committed:
Use case type constants

NoahCarnahan committed:
Add get_person_case_from_voucher to get_person_case

NoahCarnahan committed:
Make get_prescription_vouchers_from_episode more strict

NoahCarnahan committed:
Get dynamic_case_properties once

NoahCarnahan committed:
Make some triggers more strict

NoahCarnahan committed:
Check expensive condition last

NoahCarnahan committed:
Fix test case structure initialization

NoahCarnahan committed:
Update constants

NoahCarnahan committed:
Fix get_person_case()

NoahCarnahan committed:
Break voucher repeater into two

NoahCarnahan committed:
Don't forward cases from test locations

NoahCarnahan committed:
[ci skip] lint

NoahCarnahan committed:
[ci skip] lint

NoahCarnahan committed:
Revert changes to shared repeater case validation functions

NoahCarnahan committed:
Fix case validation again

NoahCarnahan committed:
actually fix it this time :(

NoahCarnahan committed:
Add 'enrolled in private sector' checks

esoergel committed:
This wasn't working

esoergel committed:
Working UserRepeater

esoergel committed:
Stick BETS in front of everything

esoergel committed:
Add location repeater

dimagimon commented on custom/enikshay/integrations/bets/models.py:
```diff
@@ -0,0 +1,2 @@
+# Something bets.repeater_generators must be imported to allow repeater generators to be registered
+from custom.enikshay.integrations.bets import repeater_generators
```
F401 'repeater_generators' imported but unused

NoahCarnahan committed:
Fix spelling typo

NoahCarnahan committed:
Merge branch 'master' into nc/bets

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into nc/bets

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into nc/bets

esoergel committed:
Is this how to handle the raw arg?

@snopoke I'm adding a receiver for SQLLocation post_save.  The test
TestSQLDumpLoad.test_location fails complaining that my receiver doesn't
anticipate a "raw" arg.  I found a couple instances of that and it looks like
essentially that bypasses the signal.  I don't know the intended use-case for
this, exactly, but could you confirm that this is the behavior you'd expect?
